multiqc/modules/sambamba/markdup.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+from collections import OrderedDict
+import os
+import logging
+import re
+
+from multiqc import config
+from multiqc.plots import bargraph
+
+log = logging.getLogger(__name__)
+
+class SambambaMarkdupMixin:
+
+    """ Find and parse Sambamba Markdup output log files and calculate duplication rate """
+
+    def parse_sambamba_markdup(self):
+
+        # Clean sample name from 'markdup_sample_1' to 'sample_1'
+        # Find and load sambamba logs to markdup_data_dict.
+        # Regex for key phrases and calculate duplicate rate.
+        # Give user warning if redundant samples are found.
+
+        self.markdup_data_dict = dict()
+
+        for f in self.find_log_files("sambamba/markdup"):
+
+            sample_name = self.markdup_clean_s_name(f["s_name"])
+            self.markdup_data_dict[sample_name] = self.get_markdup_stats(f["f"])
+
+            self.add_data_source(f)
+
+            if f["s_name"] in self.markdup_data_dict:
+                log.debug("Duplicate sample name found! Overwriting: {}".format(f["s_name"]))
+
+        if len(self.markdup_data_dict) == 0:
+            raise UserWarning

```

Comment:
```suggestion
            return 0
```

Reply:
I _think_ this may fix the tests. They're failing currently because you're generating a report when there should not be one.

Reply:
Ah it's not this sorry (though I think you do want to return `0` anyway, else other submodules won't run in the future).

It's because you're not filtering sample names: https://multiqc.info/docs/#filtering-by-parsed-sample-names

Reply:
Agreed, I added this so sambamba markdup can also honor the `--ignore-samples` flag from the user.